podman-desktop: 1.22.1 -> 1.23.1, update fetcherVersion

kachick opened the pull request:
<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

Dif: https://github.com/podman-desktop/podman-desktop/compare/v1.22.1...v1.23.1

ref: https://github.com/NixOS/nixpkgs/pull/462421#issuecomment-3550496788, https://github.com/NixOS/nixpkgs/pull/361460#discussion_r1896540669, https://github.com/NixOS/nixpkgs/pull/407982#issuecomment-2888429972

This PR includes and supersedes #462421

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform:
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- Tested, as applicable:
  - [ ] [NixOS tests] in [nixos/tests].
  - [ ] [Package tests] at `passthru.tests`.
  - [ ] Tests in [lib/tests] or [pkgs/test] for functions and "core" functionality.
- [x] Ran `nixpkgs-review` on this PR. See [nixpkgs-review usage].
- [x] Tested basic functionality of all binary files, usually in `./result/bin/`.
- Nixpkgs Release Notes
  - [ ] Package update: when the change is major or breaking.
- NixOS Release Notes
  - [ ] Module addition: when adding a new NixOS module.
  - [ ] Module update: when the change is significant.
- [x] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and other READMEs.

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

cc: @booxter 

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

r-ryantm committed:
podman-desktop: 1.22.1 -> 1.23.1

kachick committed:
podman-desktop: update fetcherVersion in pnpm.fetchDeps

The hash appears flaky when building on GitHub-hosted Linux runners.
As far as I see, applying the new fetcher introduced in 0eec779fcc1fcc027a0d37ed3b6cd251343c22a5
makes the result stable.

booxter commented:
Thanks. I can confirm the version = 2 fixes the issue, and the hash is correct.

---

But this bump seems to break Darwin. I get this error when starting the app:

```
-------------------------------------
Translated Report (Full Report Below)
-------------------------------------
Process:             .Podman Desktop-wrapped [39613]
Path:                /Volumes/VOLUME/*/Podman Desktop.app/Contents/MacOS/.Podman Desktop-wrapped
Identifier:          .Podman Desktop-wrapped
Version:             ???
Code Type:           ARM-64 (Native)
Role:                Foreground
Parent Process:      launchd [1]
Coalition:           io.podmandesktop.PodmanDesktop [37499]
User ID:             501

Date/Time:           2025-11-18 22:29:08.8288 -0500
Launch Time:         2025-11-18 22:29:08.2039 -0500
Hardware Model:      Mac16,10
OS Version:          macOS 26.0.1 (25A362)
Release Type:        User

Crash Reporter Key:  1F83FC7E-79D0-6E3E-3CC4-BC2DCF710B9E
Incident Identifier: 019DBE15-BF0F-4848-84B2-9909156C00BF

Time Awake Since Boot: 960000 seconds

System Integrity Protection: enabled

Triggered by Thread: 0

Exception Type:    EXC_BREAKPOINT (SIGTRAP)
Exception Codes:   0x0000000000000001, 0x0000000111548140

Termination Reason:  Namespace SIGNAL, Code 5, Trace/BPT trap: 5
Terminating Process: exc handler [39613]


Thread 0 Crashed:
0   Electron Framework            	       0x111548140 v8::V8::EnableWebAssemblyTrapHandler(bool) + 100
1   Electron Framework            	       0x11244b50c v8::CpuProfileNode::GetNodeId() const + 162012
2   Electron Framework            	       0x111ccd4f4 rust_png$cxxbridge1$Reader$try_get_chrm + 187688
3   Electron Framework            	       0x113ca068c _v8_internal_Node_Print(void*) + 5989516
4   Electron Framework            	       0x1119fbe24 node::PrincipalRealm::wasm_streaming_object_constructor() const + 54672
5   Electron Framework            	       0x1119fbd4c node::PrincipalRealm::wasm_streaming_object_constructor() const + 54456
6   Electron Framework            	       0x11160e690 node::inspector::protocol::Value::AppendSerialized(std::__Cr::vector<unsigned char, std::__Cr::allocator<unsigned char>>*) const + 71688
7   Electron Framework            	       0x11160b014 node::inspector::protocol::Value::AppendSerialized(std::__Cr::vector<unsigned char, std::__Cr::allocator<unsigned char>>*) const + 57740
8   Electron Framework            	       0x1118963bc v8::Message::GetScriptOrigin() const + 22332
9   Electron Framework            	       0x111895d7c v8::Message::GetScriptOrigin() const + 20732
10  Electron Framework            	       0x11235caec ElectronMain + 124
11  dyld                          	       0x181fadd54 start + 7184
```

I will check if Linux is also broken in a few minutes and update.

booxter commented:
My reading of v8 code suggests that the wasm trap handler was set twice, which is forbidden:

```
bool g_is_trap_handler_enabled{false};
std::atomic<bool> g_can_enable_trap_handler{true};

bool EnableTrapHandler(bool use_v8_handler) {
  // We should only enable the trap handler once, and before any call to
  // {IsTrapHandlerEnabled}. Enabling the trap handler late can lead to problems
  // because code or objects might have been generated under the assumption that
  // trap handlers are disabled.
  bool can_enable =
      g_can_enable_trap_handler.exchange(false, std::memory_order_relaxed);
  // EnableTrapHandler called twice, or after IsTrapHandlerEnabled.
  TH_CHECK(can_enable);
  if (!V8_TRAP_HANDLER_SUPPORTED) {
    return false;
  }

  // "Warm-up" the TrapHandlerGuard mechanism to ensure that if any
  // initialization is required for its thread-local storage, it is done now
  // and not inside the signal handler. We're being extra cautious here, it's
  // unclear if this is really necessary.
  TrapHandlerGuard active_guard;

  if (use_v8_handler) {
    g_is_trap_handler_enabled = RegisterDefaultTrapHandler();
    return g_is_trap_handler_enabled;
  }
  g_is_trap_handler_enabled = true;
  return true;
}
```

We are aborted by `TH_CHECK` call that validates that this function is called once with `true`.

booxter commented:
I think we hit https://github.com/electron/electron/pull/48975

kachick commented:
Thank you for your investigation about the root cause!
I couldn't have found this issue in such a short time. :pray:

booxter commented:
@kachick why closing it though? I'm going to post a bump request for `electron_39` in a few. Once we merge the electron bump, we can land this PR.

booxter commented:
In case you are curious, there are some details about what exactly happens (in some crashing apps) here: https://github.com/electron/electron/pull/48983

kachick committed:
podman-desktop: enforce XDG_SESSION_TYPE=x11

refs:
https://github.com/podman-desktop/podman-desktop/commit/a46458fdfd6c4ec2b0d9b01ca5e1457401c5083b
https://github.com/podman-desktop/podman-desktop/commit/d32b613358a06afe453ce74320f64fe73c1c4554

booxter commented:
> I am unsure if this is related to issue https://github.com/podman-desktop/podman-desktop/issues/14123, since I could launch the dashboard on Wayland with current 1.22.1 + electron_38.

This is not true for me. If I run it from `c8d4dabc4357a22d1c249a9363998bdb00122544` that was the last version we built from electron_37, then the window is popping up (blurred!) while with current version (built with electron 38) it doesn't.

Which makes me think that perhaps the dashboard missing issue is not specific to this bump - and perhaps better handled separately from this PR. I'd drop the x11 patch and land the version bump for now. WDYT?

booxter commented:
@NixOS/nixpkgs-merge-bot merge

kachick commented:
Thank you all for reviewing and merging!
